src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -341,4 +342,45 @@ public static void validateArraySchema(Schema arraySchema, String name) {
                                                        name, componentSchema));
     }
   }
+
+  public static void validateColumnForPartition(String columnName, Schema outputSchema) {
+    if (columnName == null) {
+      return;
+    }
+    if (outputSchema == null) {
+      throw new InvalidStageException(String.format("Can't validate column '%s' because output schema not specified",
+                                                       columnName));
+    }
+    for (Schema.Field field : outputSchema.getFields()) {
+      if (field.getName().equals(columnName)) {
+        if (!field.getSchema().getType().equals(Schema.Type.UNION)) {
```

Comment:
this can be simplified to:

```
Schema fieldSchema = field.getSchema();
fieldSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable() : fieldSchema;
Schema.LogicalType logicalType = fieldSchema.getLogicalType();
if (logicalType != ...) {
  throw ...
}
```

Reply:
Done.